radio/src/gui/colorlcd/widgets/modelbmp.cpp

```diff
@@ -99,13 +113,19 @@ class ModelBitmapWidget: public Widget
         }
 
         if (rect.h >= 96 && rect.w >= 120) {
-          buffer->drawScaledBitmap(bitmap.get(), 0, 38, width(), height() - 38);
+          buffer->drawScaledBitmap(bitmap.get(), 0, 0, width(), height() - 38);
         } else {
           buffer->drawScaledBitmap(bitmap.get(), 0, 0, width(), height());
         }
       }
     }
 };
 
-BaseWidgetFactory<ModelBitmapWidget> modelBitmapWidget("ModelBmp", nullptr);
-const WidgetFactory * defaultWidget = &modelBitmapWidget;
+const ZoneOption ModelBitmapWidget::options[] = {
+    {STR_COLOR, ZoneOption::Color, OPTION_VALUE_UNSIGNED(DEFAULT_COLOR_INDEX)},
```

Comment:
You need the real RGB565, which is an issue, as it cannot be in a static struct as `DEFAULT_COLOR` is actually a table lookup. Also, it seems to me that you store an index here, and use that later as a RGB565, which clearly is an issue.

Reply:
Hm... that's probaly what the >>16 I wasn't understanding from one of the widgets was about then?

Reply:
Hm... that's probaly what the >>16 I wasn't understanding from one of the widgets was about then?

Reply:
This is because the "constant" (it's not constant) `DEFAULT_COLOR` is defined with something like:
```
(palette[DEFAULT_COLOR_INDEX] << 16)
```
So that you need to shift the bits back to the right to have the normal color. The shift is used so you can use that value directly ORed with other flags (font size, etc) as part of `LcdFlags`.